### DISCUSSION TITLE
Creating a dashboard from scratch via serve?

### DISCUSSION THREAD

**Discussion by gespi1** [grafana/grizzly#488] [Q&A] [answered]
Hi,

Been using `grr serve` as a way to modify existing dashboards via the local grafana grr provides. However when going to create a new dashboard and save it, i receive an error stating `Failed to save dashboard`. Looking closer at the browser console the root error is `Dashboard has no UID`. Even when setting a `uid` on the json via the dashboard settings the error persists. How are users suppose create/init dashboards  from scratch? The [docs](https://grafana.github.io/grizzly/grafana/#dashboards) state you should copy paste the json of an existing dashboard from a grafana down into your local fs but this is counter intuitive because we are using grr to make modifications to our dashboards before commiting them to our version control system. Why not the creation as well? 

I would expect that creating a dashboard on the local grr grafana would generate a json|yaml on the directory specified on the  `grr serve` command

**Comment by malcolmholmes** [comment]
We haven't really covered the new dashboard case yet. But thinking about it, perhaps it wouldn't be too hard to handle.

We could have a simple form within the Grizzly server that takes in a title, UID and filesystem path. On submit, it would write the most basic file to disk at that level, and proceed normally from there.

**Comment by malcolmholmes** [comment, marked as answer]
Here's a first naive attempt at adding a "new" function. It works, although I haven't yet seen my changes in the saved file, hmm. 

https://github.com/grafana/grizzly/pull/489

**Reply by malcolmholmes** [reply]
saving is fixed. And I've styled it somewhat. Let me know what you think.

**Reply by gespi1** [reply]
this is great @malcolmholmes! Thanks on getting on it so fast. LGTM
